### ISSUE TITLE
Does Ktor support application/json type?

### ISSUE THREAD

**Issue by KennethanCeyer** [Kotlin/ktor#102]
Hello there!

Does Ktor support `application/json`  request type?

**When request is**

```bash
$ curl -H "Content-Type: application/json" -X POST -d '\
{\
  "hello: "world",\
  "content: {\
    "name": "kenneth"\
  }\
}'
http://localhost:8000
```

I'm expected as

```kotlin
data class ContentModel(
    val name: String
)

@location("/test")
data class Test(val hello: String, val content: ContentModel)

fun Route.athena() {
    location<Test> {
        method(HttpMethod.Post) {
            handle<Test> {
                // [Expected]
                // it.hello => "world"
                // it.content.name => "kenneth"
            }
        }
    }
}
```

Thank you.

**Comment by KennethanCeyer** [comment]
I think you don't have any method when the Content-Type is `application/json`.

https://github.com/Kotlin/ktor/blob/master/ktor-core/src/org/jetbrains/ktor/request/RequestContent.kt#L12

Do you have any plan to update those features or any supporting for contributors?

(Documentations or Communications)

Thank you.

**Comment by orangy** [comment]
This is fixed in 0.4.0 by introducing `call.receive<MyClass>` after installing `GsonSupport` feature. Will be improved in next version with proper extensible content-negotiation feature.
